### PR TITLE
[dev-v2.9] feat: bump harvester-cloud-provider to v0.2.4

### PR DESCRIPTION
**Pull request by FrankYang0529**
## Issue: 

https://github.com/harvester/harvester/issues/4947
https://github.com/harvester/harvester/issues/5247
https://github.com/harvester/charts/pull/230

### CODE REVIEW

**Review by w13915984028 — APPROVED** [review]
LGTM, thanks.

File: release.yaml [hunk 1]
```diff
@@ -30,9 +30,10 @@ longhorn-crd:
   - 102.3.3+up1.5.5
   - 103.2.3+up1.5.5
 harvester-cloud-provider:
-  - 104.0.0+up0.2.3
   - 102.0.2+up0.2.3
   - 103.0.1+up0.2.3
+  - 104.0.0+up0.2.3
```
Comment (lucasmlp) [inline release.yaml]: Both these versions will be released in `v2.9.0`?

Comment (FrankYang0529) [reply release.yaml]: @w13915984028, I think we can remove `102.0.2+up0.2.3` and `103.0.1+up0.2.3`. WDYT? Thank you.

Comment (w13915984028) [reply release.yaml]: Need to hear advise from @lucasmlp, thanks.

Normally, a Chart for old Rancher version, e.g. `102.0.2+up0.2.3` for Rancher v2.7, should/not also be visible&available on new Rancher version v2.9?

Our past PRs were adding a new Harvester related chart to all mainstream Rancher version v2.7, v2.8, v2.9, but not removing any existing chart versions.

Comment (w13915984028) [reply release.yaml]: And according to the naming rules, we already have https://github.com/rancher/charts/blob/dev-v2.9/charts/harvester-cloud-provider/104.0.0%2Bup0.2.3/values.yaml,  

This PR seems should name it as `104.0.1+up0.2.3`. @FrankYang0529

File: index.yaml [hunk 2]
```diff
@@ -1855,6 +1855,36 @@ entries:
...
+      catalog.cattle.io/release-name: harvester-cloud-provider
+      catalog.cattle.io/ui-component: harvester-cloud-provider
+      catalog.cattle.io/upstream-version: 0.2.4
+    apiVersion: v2
+    appVersion: v0.2.1
+    created: "2024-05-07T18:23:43.582162+08:00"
+    dependencies:
+    - name: kube-vip
+      repository: file://./charts/kube-vip
+      version: 0.4.2
```
Comment (w13915984028) [inline index.yaml]: @FrankYang0529 Comparing to another 2 PRs, `0.4.2` is still existing in index.yaml;

it is due to `Chart.yaml.patch` has no `-` line to remove this version, please double check, thanks.

PR for dev-2.8 removed it.
image

**Review by lucasmlp — APPROVED** [review]
LGTM 🚀

**Pull request merged**